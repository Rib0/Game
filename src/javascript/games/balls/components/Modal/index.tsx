import React from 'react';
import styled from 'styled-components';

import Button from '../Button';

interface IModalProps {
    isActive: Boolean;
    header: string;
    content: React.ReactNode;
    acceptButtonText: string;
    acceptButtonCallback: (e: React.MouseEvent<HTMLButtonElement>) => void;
    declineButtonText?: string;
    declineButtonCallback?: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const Overlay = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background: gray;
    opacity: 0.5;
    z-index: 98;
    pointer-events: none;
`;

const ModalContainer = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 20px;
    z-index: 99;
    border-radius: 5px;
    width: 40%;
    background-color: white;
    box-shadow: inset 0 0 10px gray;
`;

const ModalHeader = styled.div`
    margin-bottom: 20px;
    text-align: center;
    font-weight: bold;
`;

const ModalContent = styled.div`
    margin-bottom: 20px;
    text-align: center;
`;

const ModalButtons = styled.div`
    text-align: center;

    ${Button}:not(:last-child) {
        margin-right: 10px;
    }
`;

const Modal = ({
    isActive,
    header,
    content,
    acceptButtonText,
    acceptButtonCallback,
    declineButtonText,
    declineButtonCallback,
}: IModalProps) => {
    if (!isActive) return null;

    return (
        <>
            <Overlay />
            <ModalContainer>
                <ModalHeader>{header}</ModalHeader>
                <ModalContent>{content}</ModalContent>
                <ModalButtons>
                    <Button text={acceptButtonText} onClick={acceptButtonCallback} />
                    {declineButtonText && (
                        <Button text={declineButtonText} onClick={declineButtonCallback} />
                    )}
                </ModalButtons>
            </ModalContainer>
        </>
    );
};

export default Modal;
