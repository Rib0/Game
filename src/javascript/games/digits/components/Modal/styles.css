.overlay {
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background: var(--colorBlack);
    opacity: 0.5;
}

.modal {
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 15px;
    border: 2px solid red;
    border-radius: 10px;
    z-index: 99;
    background: var(--colorWhite);
    width: 33%;
    text-align: center;
}

.caption {
    font-weight: bold;
}

.content {
    margin-top: 10px;
}

.buttons {
    margin-top: 10px;
}

.button {
    color: red;
    border-color: red;
    padding: 0 20px;
}
