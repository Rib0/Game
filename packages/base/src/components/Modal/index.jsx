import React from 'react';

import Button from '@/components/Button';

import styles from './styles.css';

export default (props) => {
    const { caption, isOpen, children, onAccept, acceptButtonText } = props;

    if (!isOpen) return null;

    return (
        <>
            <div className={styles.overlay} />
            <div className={styles.modal}>
                {caption && <div className={styles.caption}>{caption}</div>}
                <div className={styles.content}>
                    {children}    
                </div>
                <div className={styles.buttons}>
                    <Button type="styled" text={acceptButtonText || 'ะะบ'} onClick={onAccept} />
                </div>
            </div>
        </>
    )
}