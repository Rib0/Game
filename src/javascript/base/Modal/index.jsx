import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Button } from 'Base';

import styles from './styles.css';

class Modal extends PureComponent {
    static Content({ children }) {
        return <div className={styles.content}>{children}</div>;
    }

    render() {
        const { caption, isOpen, children, onAccept, acceptButtonText } = this.props;

        if (!isOpen) return null;

        return (
            <>
                <div className={styles.overlay} />
                <div className={styles.modal}>
                    {caption && <div className={styles.caption}>{caption}</div>}
                    {children}
                    <div className={styles.buttons}>
                        <Button text={acceptButtonText || 'ะะบ'} onClick={onAccept} />
                    </div>
                </div>
            </>
        );
    }
}

Modal.propTypes = {
    caption: PropTypes.string,
    isOpen: PropTypes.bool,
    children: PropTypes.element,
    onAccept: PropTypes.func,
    acceptButtonText: PropTypes.string,
};

export default Modal;
