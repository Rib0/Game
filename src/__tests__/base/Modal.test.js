import React from 'react';
import Modal from 'base/Modal';

const setUp = props => mount(<Modal {...props} />);

describe('Modal test', () => {
    const props = {
        caption: 'some caption',
        isOpen: true,
        onAccept: jest.fn(),
        acceptButtonText: 'Ok',
    };

    test('Modal should render', () => {
        const Component = setUp(props);
        expect(Component).toMatchSnapshot();
    });
    test('Modal should render component', () => {
        const Component = setUp(props);
        expect(Component.find('Modal')).toHaveLength(1);
    });
    test('Modal should render null', () => {
        const newProps = {
            ...props,
            isOpen: false,
        };

        const Component = setUp(newProps);
        expect(Component.isEmptyRender()).toBeTruthy();
    });
    test('Modal should render caption', () => {
        const Component = setUp(props);
        expect(Component.find('div.caption').text()).toBe('some caption');
    });
    test('Modal should render custom accept button text', () => {
        const newProps = {
            ...props,
            acceptButtonText: undefined,
        };

        const Component = setUp(newProps);
        expect(Component.find('Button').prop('text')).toBe('ะะบ');
    });
    test('Modal should render button with text', () => {
        const Component = setUp(props);
        expect(Component.find('Button').text()).toBe('Ok');
    });
    test('onAccept callback should call if button pressed', () => {
        const Component = setUp(props);
        const button = Component.find('Button');
        button.simulate('click');
        expect(props.onAccept).toHaveBeenCalledTimes(1);
    });
    test('Modal static content method should render', () => {
        const Component = shallow(
            <Modal.Content>
                <div />
                <div />
            </Modal.Content>
        );
        const divs = Component.find('div');
        expect(divs).toHaveLength(3);
    });
});
