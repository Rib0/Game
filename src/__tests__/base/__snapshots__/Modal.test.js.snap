// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test Modal should render 1`] = `
<Modal
  acceptButtonText="Ok"
  caption="some caption"
  isOpen={true}
  onAccept={[MockFunction]}
>
  <div
    className="overlay"
  />
  <div
    className="modal"
  >
    <div
      className="caption"
    >
      some caption
    </div>
    <div
      className="buttons"
    >
      <Button
        onClick={[MockFunction]}
        size="medium"
        text="Ok"
        type="styled"
      >
        <button
          className="button styled medium"
          onClick={[MockFunction]}
          type="button"
        >
          Ok
        </button>
      </Button>
    </div>
  </div>
</Modal>
`;
